be/src/exec/pipeline/adaptive/event.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <atomic>
+#include <memory>
+#include <vector>
+
+#include "exec/pipeline/adaptive/adaptive_fwd.h"
+#include "exec/pipeline/pipeline_fwd.h"
+#include "runtime/runtime_state.h"
+
+namespace starrocks::pipeline {
+
+/// Event is used to drive the execution of pipelines.
+/// Event::schedule is invoked when its dependencies are all finished.
+class Event : public std::enable_shared_from_this<Event> {
+public:
+    Event() = default;
+    virtual ~Event() = default;
+
+    /// This is not thread-safe.
+    virtual void schedule(RuntimeState* state) {}
```

Comment:
What's schedule meaning of the event itself? maybe handle、process、dispatch?

Reply:
`schedule` means this event will schedule some pipeline drivers to execute. `process` is better.